### ISSUE TITLE
Configuration based on platform name

### ISSUE THREAD

**Issue by srid** [syl20bnr/spacemacs#2528]
Does spacemacs provide any abstraction to set a config value based on platform? 

For example I want the `dotspacemacs-default-font` font name value to be different on OSX and Windows (profont uses different font name on both platforms) without having to write the verbose [if-then-else elisp code](http://stackoverflow.com/questions/26204761/programmatically-distinguishing-system-type-windos-xp-v-windows-7) myself (for keeping .spacemacs simple)

**Comment by srid** [comment]
Just found this by grep'ing the source code:

```elisp
(if (system-is-mswindows) "ProFontWindows" "ProFontX")
```

**Comment by travisbhartwell** [comment]
I've added for my own personal use the ability to have hostname specific configurations.  It would be pretty trivial to add something similar for platform-based things.  Perhaps my strategy could be abstracted and folded into Spacemacs proper.

First, see this:
https://gitlab.com/travisbhartwell/vcsh_emacs/blob/675d2d3b20d6ee7c232bbcd7d2e8e21483fc51a3/.spacemacs.d/init.el#L22

There I load a file in `~/.spacemacs.d` that has a name corresponding to the hostname, if any.

Then, at the end of the `dotspacemacs/layers`, `dotspacemacs/init`, and `dotspacemacs/config` functions I do something similar to this:
https://gitlab.com/travisbhartwell/vcsh_emacs/blob/675d2d3b20d6ee7c232bbcd7d2e8e21483fc51a3/.spacemacs.d/init.el#L88

If a corresponding function named `tbh/dotspacemacs/layers` (tbh are my initials) has been defined, it will call that function.  For example, for `shedemei`, my desktop machine, I have this:

https://gitlab.com/travisbhartwell/vcsh_emacs/blob/675d2d3b20d6ee7c232bbcd7d2e8e21483fc51a3/.spacemacs.d/shedemei.el#L2

Basically in each of these functions, I override or add to the stuff that has been defined as a default in `~/.spacemacs.d/init.el`.

I certainly could see this extended to include platform specific stuff as well.  Does this make sense to add to Spacemacs proper?

**Comment by srid** [comment]
That is interesting. It would be nice if spacemacs included this abstraction mechanism of multiplexing over both hostname and system-type. FWIW, my current setup [is this](https://github.com/srid/home/commit/a03b1db433477a3f702a6dc79e5a1edc808365c7) and I may well evolve it to something similar to what @travisbhartwell has.

**Comment by travisbhartwell** [comment]
cc'ing @syl20bnr to make sure he sees this, even though the issue is closed.

What do you think?  With guidance, I could add this to Spacemacs proper.
